src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphEntityDAO.scala

```diff
@@ -100,6 +107,9 @@ class GraphEntityDAO extends EntityDAO with GraphDAO {
   }
 
   override def cloneEntities(destWorkspaceContext: WorkspaceContext, entities: Seq[Entity], txn: RawlsTransaction) = txn withGraph { db =>
-    entities.map( e => save(destWorkspaceContext, e, txn) )
+    //save the entities with no references to other entities first
```

Comment:
(so I think all you need is something that says "this makes sure that a vertex for each entity exists before we try to reference it; since references are preserved in `saveSubObject`, we can correctly save entities with reference cycles)